tools-v2/pkg/cli/command/curvefs/warmup/list/list.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/pkg/xattr"
+	"github.com/spf13/cobra"
+)
+
+const (
+	listExample = `$ curve fs warmup list `
+
+	CURVEFS_WARMUP_OP_XATTR = "curvefs.warmup.op.list"
+)
+
+type ListCommand struct {
+	basecmd.FinalCurveCmd
+	MountpointInfo *mountinfo.MountInfo
+	mountpoint     string
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ListCommand)(nil) // check interface
+
+func NewListWarmupCommand() *ListCommand {
+	lCmd := &ListCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "list",
+			Short:   "list the ongoing warmup progress for all tasks that are currently in process",
+			Example: listExample,
+		},
+	}
+	basecmd.NewFinalCurveCli(&lCmd.FinalCurveCmd, lCmd)
+	return lCmd
+}
+
+func NewListCommand() *cobra.Command {
+	return NewListWarmupCommand().Cmd
+}
+
+func (lCmd *ListCommand) AddFlags() {
+	config.AddIntervalOptionFlag(lCmd.Cmd)
+}
+
+func (lCmd *ListCommand) Init(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return errors.New("please provide the path where CurveFS is being mounted. i.e., curve fs warmup list <curvefs mounted path>")
+	}
+
+	mountpoints, err := cobrautil.GetCurveFSMountPoints()
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	} else if len(mountpoints) == 0 {
+		return errors.New("no curvefs mountpoint found")
+	}
+
+	absPath, _ := filepath.Abs(args[0])
+	lCmd.mountpoint = absPath
+
+	lCmd.MountpointInfo = nil
+	for _, mountpointInfo := range mountpoints {
+		rel, err := filepath.Rel(mountpointInfo.MountPoint, absPath)
+		if err == nil && !strings.HasPrefix(rel, "..") {
+			// found the mountpoint
+			if lCmd.MountpointInfo == nil ||
+				len(lCmd.MountpointInfo.MountPoint) < len(mountpointInfo.MountPoint) {
+				// Prevent the curvefs directory from being mounted under the curvefs directory
+				// /a/b/c:
+				// test-1 mount in /a
+				// test-1 mount in /a/b
+				// warmup /a/b/c.
+				lCmd.MountpointInfo = mountpointInfo
+			}
+		}
+	}
+	return nil
+}
+
+func (lCmd *ListCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&lCmd.FinalCurveCmd, lCmd)
+}
+
+func (lCmd *ListCommand) RunCommand(cmd *cobra.Command, args []string) error {
+
+	resultBytes, err := xattr.Get(lCmd.mountpoint, CURVEFS_WARMUP_OP_XATTR)
+	resultStr := string(resultBytes)
+
+	if err != nil {
+		return err
+	}
+
+	entries := strings.Split(resultStr, ";")
+
+	tableRows := make([]map[string]string, 0)
+
+	if resultStr == "finished" {
+		row := make(map[string]string)
+		lCmd.Header = append(lCmd.Header, cobrautil.ROW_RESULT)
+		row[cobrautil.ROW_RESULT] = "No tasks in queue"
+		tableRows = append(tableRows, row)
+	} else {
+		lCmd.Header = append(lCmd.Header, cobrautil.ROW_PATH)
+		lCmd.Header = append(lCmd.Header, cobrautil.ROW_PROGRESS)
+		rows := make([]map[string]string, 0)
+		for _, entry := range entries {
+			if entry == "" {
+				continue
+			}
+	
+			parts := strings.Split(entry, ":")
+	
+			if len(parts) != 2 {
+				fmt.Println("Invalid entry:", entry)
```

Comment:
Try to return an error instead of printing midway

Reply:
done